app/src/main/java/com/github/steroidteam/todolist/viewmodel/NoteSelectionViewModel.java

```diff
@@ -0,0 +1,67 @@
+package com.github.steroidteam.todolist.viewmodel;
+
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.MutableLiveData;
+import androidx.lifecycle.ViewModel;
+import com.github.steroidteam.todolist.database.Database;
+import com.github.steroidteam.todolist.database.DatabaseFactory;
+import com.github.steroidteam.todolist.model.notes.Note;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+public class NoteSelectionViewModel extends ViewModel {
+
+    private final Database database;
+    private final MutableLiveData<List<Note>> noteList;
+
+    public NoteSelectionViewModel() {
+        super();
+        this.database = DatabaseFactory.getDb();
+        this.noteList = new MutableLiveData<>(new ArrayList<>());
+        this.database.getNotesList().thenAccept(this::setArrayOfNote);
+    }
+
+    public LiveData<List<Note>> getNoteList() {
+        return this.noteList;
+    }
+
+    private void setArrayOfNote(List<UUID> noteIDList) {
+        ArrayList<Note> privateArrayList = new ArrayList<>();
+        if (noteIDList.size() == 0) {
+            noteList.setValue(privateArrayList);
+        } else {
+            for (int i = 0; i < noteIDList.size(); i++) {
+                this.database
+                        .getNote(noteIDList.get(i))
+                        .thenAccept(
+                                note -> {
+                                    privateArrayList.add(note);
```

Comment:
I wonder if this is thread-safe. Notes are fetched concurrently and are added out-of-order in the list. Maybe something to address later.